### DISCUSSION TITLE
Authentication extensions?

### DISCUSSION THREAD

**Discussion by hylobates-agilis** [FreshRSS/FreshRSS#6574] [Q&A] [answered]
If I ever wanted to write an authentication extension, would that be even possible? From what I read, I fell like the extensions are more targeted towards modifying the UX/UI but not necessarily the root functionality (like authentication).

Am I wrong? If yes, can you point me to the corresponding documentation or an example documentation?

**Comment by Frenzie** [comment]
I think you're wrong about your feeling, but that's neither here nor there. We can always discuss adding another hook if necessary, but you'd have to be a little bit more concrete. You might be looking for `simplepie_before_init`.
https://github.com/FreshRSS/FreshRSS/blob/7a9a24a0d0f2e5cc674995ce3174116803acde09/lib/Minz/ExtensionManager.php#L24-L93

**Reply by hylobates-agilis** [reply]
From what I understand, `simplepie_before_init` is triggered after authentication. I want to built a new authentication extension, which might not be possible for now (maybe this needs a new hook…).

**Reply by Alkarex** [reply]
`simplepie_before_init` can be used to modify the `SimplePie` instance *before* doing any request.
But as @Frenzie is writing, we can add more hooks as needed

**Comment by Alkarex** [comment]
For user authentication to the UI, since we are supporting HTTP Basic Auth, it allows building other mechanisms on top. For instance, our support of OpenID Connect is taking advantage of that https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect.html

**Reply by hylobates-agilis** [reply]
Yep, but that mechanisms on top is external to FreshRSS, right?

Imagine I want to add LDAP authentication: either I’m building it inside FreshRSS as extension (which was my first intention), or manipulate my presentation/HTTP server to do so. Am I correct?

**Reply by Alkarex** [reply, marked as answer]
Indeed, just like our OpenID Connect solution, which is on top of FreshRSS by taking advantage of https://github.com/OpenIDC/mod_auth_openidc

**Reply by hylobates-agilis** [reply]
So I’m gonna try to hack it inside then, just to try to have it to work properly. Thanks for the answer!
